runtime/westend/src/lib.rs

```diff
@@ -1030,6 +1029,75 @@ sp_api::impl_runtime_apis! {
 		}
 	}
 
+	#[cfg(feature = "try-runtime")]
+	impl frame_try_runtime::TryRuntime<Block> for Runtime {
+		fn on_runtime_upgrade(target: frame_try_runtime::Target) -> Weight {
+			frame_support::debug::RuntimeLogger::init();
+
+			let weight = match target {
+				frame_try_runtime::Target::All => {
+					frame_support::debug::info!("Dry-running all on-runtime-upgrades.");
+					Executive::try_runtime_upgrade()
+				},
+				frame_try_runtime::Target::Pallet(name) => {
+					let name = sp_std::str::from_utf8(&name).unwrap();
+					frame_support::debug::info!("Dry-running on-runtime-upgrade of {}.", name);
+
+					frame_try_runtime::match_pallet_on_runtime_upgrade!(name,
+						System,
```

Comment:
Can we use `AllModules` rather than writing these all manually?

Reply:
Easily, no, because `AllModules` is a tuple, and tuples are by definition heterogeneous and not enumerable. What would have worked was to also create a type alias in `construct_runtime!` called `AllModulesArray`, which has the type `[Box<dyn PalletCommonTrait>; <number-of-pallets>]` (or just `Vec` for simplicity, ain't matter), where `PalletCommonTrait` is a wrapper trait for all the common stuff that all `Pallet` structs implement: `trait PalletCommonTrait: OnRuntimeUpgrade + OnInitialize + OnFinalize + OffchainWorker + PalletInfo + <anything else>`. 

I think this is the best way, but I'd see it as a follow-up task. 

@thiolliere I disucssed this with you as well in DM, chime in if I am wrong please.

Reply:
I think I have a solution for it, I'm trying

Reply:
one solution could be to have another method:
```diff
diff --git a/frame/support/src/traits.rs b/frame/support/src/traits.rs
index 602bc1aa1a..d62f6340e3 100644
--- a/frame/support/src/traits.rs
+++ b/frame/support/src/traits.rs
@@ -1566,6 +1566,9 @@ pub trait OnRuntimeUpgrade {
        fn post_upgrade() -> Result<(), &'static str> {
                Ok(())
        }
+
+       #[cfg(feature = "try-runtime")]
+       fn try_individual_on_runtime_upgrade_filtered(filter: &str) -> Result<(), &'static str>;
 }
 
 #[impl_for_tuples(30)]
@@ -1589,6 +1592,13 @@ impl OnRuntimeUpgrade for Tuple {
                for_tuples!( #( result = result.and(Tuple::post_upgrade()); )* );
                result
        }
+
+       #[cfg(feature = "try-runtime")]
+       fn try_individual_on_runtime_upgrade_filtered(filter: &str) -> Result<(), &'static str> {
+               let mut result = Ok(());
+               for_tuples!( #( result = result.and(Tuple::try_individual_on_runtime_upgrade_filtered()); )* );
+               result
+       }
 }
 
 /// Off-chain computation trait.
```
And make pallet macro implementing it using PalletInfo name.

But actually I wonder the interest, why not running all runtime upgrade everytime ?
Also the order of runtime ugprades is important and some can break other.
(like if pallet A depend on pallet B, but pallet B hasn't upgraded yet.

Reply:
This has been removed altogether.